http-clients/src/main/java/com/palantir/remoting/http/errors/SerializableErrorErrorDecoder.java

```diff
@@ -101,18 +106,16 @@ public Exception decode(String methodKey, Response response) {
         }
     }
 
-    private static Exception constructException(Class<? extends Exception> exceptionClass, String message) {
-        try {
-            return exceptionClass.getConstructor(String.class).newInstance(message);
-        } catch (Exception e1) {
-            return new RuntimeException(String.format(
-                    "Failed to construction exception of type %s, constructing RuntimeException instead: %s%n%s",
-                    exceptionClass.toString(), e1.toString(), message));
+    private static Exception constructException(
```

Comment:
I don't think this change is correct. The problem I noticed is that the server side ``JsonExceptionMapper`` is capturing the web specific exceptions like 405 Not Allowed and throwing them as a 500. I think the fix needs to be in the exception mapper to throw the right status code (and we should also decode the exception).